### PR TITLE
Dịch "linear-regression-scratch_vn" - Phần 7

### LINKED ISSUE #623: Dịch "linear-regression-scratch_vn" - Phần 7
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_linear-networks/linear-regression-scratch_vn.md#L298-L346

### PR DESCRIPTION
**Pull request by thunguyenuehk39**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #623

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [ ] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [x] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 1]
```diff
@@ -321,7 +321,14 @@ Because our loss is calculated as a sum over the batch of examples, we normalize
 so that the magnitude of a typical step size does not depend heavily on our choice of the batch size.
 -->
 
-*dịch đoạn phía trên*
+Ở mỗi bước, sử dụng một batch được rút ngẫu nhiên từ mẫu, chúng ta sẽ ước tính được gradient của mất mát đối với các tham số.
```
Comment (lkhphuc) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Tại mỗi bước, sử dụng một batch được rút ngẫu nhiên từ mẫu, chúng ta sẽ ước tính được gradient của mất mát theo các tham số.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 2]
```diff
@@ -321,7 +321,14 @@ Because our loss is calculated as a sum over the batch of examples, we normalize
...
+Tiếp theo đó, chúng ta sẽ cập nhật các tham số ( một lượng nhỏ) theo chiều hướng làm giảm sự mất mát.
```
Comment (lkhphuc) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Tiếp theo đó, chúng ta sẽ cập nhật các tham số (với một lượng nhỏ) theo chiều hướng làm giảm sự mất mát.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 3]
```diff
@@ -321,7 +321,14 @@ Because our loss is calculated as a sum over the batch of examples, we normalize
...
+Nhắc lại :numref:`sec_autograd` mà sau khi chúng ta gọi ‘backward’ ở mỗi tham số (‘param’) sẽ có gradient của chúng được chứa ở `param.grad`.
```
Comment (lkhphuc) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Nhớ lại từ :numref:`sec_autograd` rằng sau khi chúng ta gọi ‘backward’, mỗi tham số (`param`) sẽ có gradient của nó lưu ở `param.grad`.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 4]
```diff
@@ -321,7 +321,14 @@ Because our loss is calculated as a sum over the batch of examples, we normalize
...
+Đoạn mã sau áp dụng cho việc cập nhật SGD, đưa ra một bộ các tham số, tốc độ học và kích cỡ batch.
+Kích cỡ của bước cập nhật được xác định bởi tốc độ học `lr`.
+Bởi vì các mất mát được tính dựa trên tổng batch của các mẫu, chúng ta bình thường hóa kích cỡ bước theo kích cỡ của batch (`batch_size`),
```
Comment (lkhphuc) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Bởi vì các mất mát được tính dựa trên tổng các mẫu của batch, chúng ta chuẩn hóa kích cỡ bước cập nhật theo kích cỡ của batch (`batch_size`),  sao cho độ lớn của một bước cập nhật thông thường không phụ thuộc nhiều vào kích cỡ batch.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 5]
```diff
@@ -321,7 +321,14 @@ Because our loss is calculated as a sum over the batch of examples, we normalize
...
+sao cho độ lớn của kích thước của bước thông thường không phụ thuộc nhiều vào việc lựa chọn kích cỡ batch.
```
Comment (lkhphuc) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
```
Mình gộp vào câu trước.

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 6]
```diff
@@ -338,14 +345,15 @@ def sgd(params, lr, batch_size):
...
-## *dịch tiêu đề phía trên*
+## Huấn luyện
 
 <!--
 Now that we have all of the parts in place, we are ready to implement the main training loop.
 It is crucial that you understand this code because you will see nearly identical training loops over and over again throughout your career in deep learning.
 -->
 
-*dịch đoạn phía trên*
+Bây giờ, sau khi đã đi qua được các phần, chúng ta đã sẵn sàng để thực hiện việc huấn luyện.
```
Comment (lkhphuc) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Bây giờ, sau khi đã có tất cả các thành phần, chúng ta đã sẵn sàng để viết vòng lặp huấn luyện.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 7]
```diff
@@ -338,14 +345,15 @@ def sgd(params, lr, batch_size):
...
+Quan trọng nhất là bạn phải hiểu được những dòng code bởi vì việc huấn luyện này được lặp lại trong suốt qua trình chúng ta tìm hiểu và thực hiện việc học sâu.
```
Comment (lkhphuc) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Quan trọng nhất là bạn phải hiểu được rõ đoạn mã này bởi vì việc huấn luyện gần như tương tự thế này sẽ được lặp lại nhiều lần trong suốt qua trình chúng ta tìm hiểu và lập trình các thuật toán học sâu.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 8]
```diff
@@ -355,7 +363,11 @@ Finally, we will call the optimization algorithm `sgd` to update the model param
 Since we previously set the batch size `batch_size` to $10$, the loss shape `l` for each minibatch is ($10$, $1$).
 -->
 
-*dịch đoạn phía trên*
+Trong mỗi vòng lặp, chúng ta sẽ lấy ra các minibatch của mô hình, đầu tiên chúng sẽ được chuyển vào mô hình của chúng ta để phục vụ cho việc thiết lập dự đoán.
```
Comment (lkhphuc) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Trong mỗi vòng lặp, đầu tiên chúng ta sẽ lấy ra các minibatch dữ liệu và chạy nó qua mô hình để lấy ra tập kết quả dự đoán.
```
Chỗ này bản gốc viết lủng củng với bị sai chỗ chỗ `minibatch of models`.

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 9]
```diff
@@ -355,7 +363,11 @@ Finally, we will call the optimization algorithm `sgd` to update the model param
...
+Sau khi tính toán sự mất mát, chúng ta dùng hàm `backward` để bắt đầu truyền ngược qua mạng lưới,
```
Comment (lkhphuc) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Sau khi tính toán sự mất mát, chúng ta dùng hàm `backward` để bắt đầu lan truyền ngược qua mạng lưới, lưu trữ các gradient tương ứng với mỗi tham số trong từng thuộc tính `.grad` của chúng.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 10]
```diff
@@ -355,7 +363,11 @@ Finally, we will call the optimization algorithm `sgd` to update the model param
...
+lưu trữ các gradient tương ứng với mỗi tham số trong từng thuộc tính của chúng `.grad`.
```
Comment (lkhphuc) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 11]
```diff
@@ -355,7 +363,11 @@ Finally, we will call the optimization algorithm `sgd` to update the model param
...
+Cuối cùng, chúng ta sẽ dùng thuật toán tối ưu `sgd` để cập nhật các tham số của mô hình.
+Từ đuầm chúng ta đã đặt kích thước batch `batch_size` là $10$, mất mát `I` cho mỗi minibatch là ($10$,$1$).
```
Comment (lkhphuc) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Từ đầu chúng ta đã đặt kích thước batch `batch_size` là $10$, vậy nên mất mát `I` cho mỗi minibatch có kích thước là ($10$,$1$).
```
